pytorch_lightning/trainer/trainer.py

```diff
@@ -2612,6 +2613,36 @@ def weights_summary(self, val: Optional[str]) -> None:
         rank_zero_deprecation("Setting `Trainer.weights_summary` is deprecated in v1.5 and will be removed in v1.7.")
         self._weights_summary = val
 
+    """
+    Estimator Properties
+    """
+
+    @property
+    def estimated_num_training_steps(self) -> int:
```

Comment:
since accumulation is factored into the value here, the name "training_steps" is not really accurate here, right? When accumulating, we do indeed perform a training step, just no optimizer step.

Reply:
how about `estimated_num_optimization_steps`?

Reply:
estimated_num_optimization_steps: I like it more than training steps. I think it is as close as we can get.